### ISSUE TITLE
Multipart email with notmuch

### ISSUE THREAD

**Issue by kausban** [jeremy-compostella/org-msg#113]
Hi, unsure if this was discussed or some config changes took place recently. I used to be able to send/reply to emails with both text and html parts. Something changed after I updated to a recent release last week. Note, I am using **notmuch**.

It seems if the property `:alternatives:` is set to either `(html text)` or `(text html)`, only a "text/html" email is sent. Setting exporter to text only or utf-8 works correctly.

These are my current settings for reference
```
(setq org-msg-options "html-postamble:nil H:5 num:nil ^:{} toc:nil author:nil email:nil \\n:t"
        org-msg-startup "hidestars indent inlineimages"
        org-msg-greeting-fmt "\nHi *%s*,\n\n"
        org-msg-greeting-name-limit 3
        org-msg-default-alternatives '((new             . (utf-8))
                                       (reply-to-html   . (utf-8 html))
                                       (reply-to-text   . (utf-8)))
        org-msg-posting-style 'top-posting
        org-msg-convert-citation t
        org-msg-signature "
#+begin_signature

#+end_signature")
```

BTW: can is there a config to make email forwards use top-posting? I believe that is the standard.

**Comment by obar** [comment]
I'm also using notmuch but don't think this has a bearing on the problem you're describing. I've got multipart emails going out properly, did you change it to `(text html)` for all the new/reply options? ie:

```
        org-msg-default-alternatives '((new             . (text html))
                                       (reply-to-html   . (text html))
                                       (reply-to-text   . (text html)))
```

I can look in my sent folder (`j t` from notmuch) and see multipart messages.

I _think_ I see what you mean about forwards behaving differently than reply with top-posting, but I'm not certain because I've got org-msg pretty heavily customized (changed some functions) and I haven't tested this thoroughly. Perhaps that should be opened in a separate issue.

**Comment by kausban** [comment]
> I'm also using notmuch but don't think this has a bearing on the problem you're describing. 
I am unsure if this is a notmuch only issue. I just tested it again and setting `(text html)` sends only an html msg (notmuch will show it is a with the following label '[text/html]'.

This used to work fine previously (perhaps a commit from last year). This change happened after I updated - I am using the current master with the last commit on May 19. Are you using the same release?

> I _think_ I see what you mean about forwards behaving differently than reply with top-posting, but I'm not certain because 

Forwards are generally top-posted. For example, if I forward a 10 email thread in my inbox right now to someone, "end of forward message", the org-msg preamble, body and signature are added after the entirety of the forwarded content on line 500. A recipient will most likely miss my msg unless they scroll all the way down. 

The default behaviour of org-msg seems to be bottom posting forwards.

> Perhaps that should be opened in a separate issue.

Probably. I assumed  I was doing something wrong, since bottom posting with forwards is non-standard.

**Comment by obar** [comment]
Yep, I'm on the most recent release, using notmuch, and not seeing this multipart glitch.

There was a change to how multipart gets configured within the last year but the new config works for me.

**Comment by kausban** [comment]
Don't know what to say. I tried a clean config and with `(text html)` the sent mail is only of the form "text/html" instead of the previous separate text and html components of a multi-part email.

I will attempt to downgrade org-msg next weekend  to find after which commit I lose the functionality.

**Comment by obar** [comment]
Can you share the config you just tried?

**Comment by exzombie** [comment]
I'm having the same issue, but on mu4e, so this does not appear to be specific to notmuch. As soon as `html` is in the list of alternatives, the mail will only contain `text/html` and `application/pgp-signature` parts, *nothing else*. Meaning, I can't send attachments, they are silently dropped.

Which further information should I provide?

**Comment by DarkBuffalo** [comment]
For my part in mu4e has the response of an HTML mail it does not even display the response whether in text or HTML format.
